src/SDK/Resource/ResourceDetectorInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Resource;
+
+interface ResourceDetectorInterface
+{
+    public function getResource(): ResourceInfo;
+}
```

Comment:
Just for reference.
We will have to change the AWS detectors accordingly: https://github.com/open-telemetry/opentelemetry-php-contrib/tree/main/src/Aws